packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -890,31 +859,27 @@ export class AngularCompilerPlugin {
 
     // If nothing we care about changed and it isn't the first run, don't do anything.
     if (changedFiles.length === 0 && !this._firstRun) {
-      return Promise.resolve();
+      return;
     }
 
     // Make a new program and load the Angular structure.
     await this._createOrUpdateProgram();
 
-    if (this.entryModule) {
-      // Try to find lazy routes if we have an entry module.
-      // We need to run the `listLazyRoutes` the first time because it also navigates libraries
-      // and other things that we might miss using the (faster) findLazyRoutesInAst.
-      // Lazy routes modules will be read with compilerHost and added to the changed files.
-      if (this._ngCompilerSupportsNewApi) {
-        this._processLazyRoutes(this._listLazyRoutesFromProgram());
-      } else if (this._firstRun) {
-        this._processLazyRoutes(this._getLazyRoutesFromNgtools());
-      } else {
-        const changedTsFiles = this._getChangedTsFiles();
-        if (changedTsFiles.length > 0) {
-          this._processLazyRoutes(this._findLazyRoutesInAst(changedTsFiles));
-        }
-      }
-      if (this._options.additionalLazyModules) {
-        this._processLazyRoutes(this._options.additionalLazyModules);
+    // Try to find lazy routes if we have an entry module.
```

Comment:
The `entryModule` check is still required as without it a number of lazy loading tests fails. Mainly the once that are using import or `System.import` as it will fail with an error

Cannot read property 'isComponent' of undefined

Reply:
It's only required with JIT mode actually.  The application metadata is only fully analyzed with AOT.  The `listLazyRoutes` method needs a starting point in JIT as a result since it doesn't have any other information to use.